libraries/utils/src/render-lines-on-sp.ts

```diff
@@ -38,13 +34,15 @@ export class ScatterPlotLinesRenderer {
         this.sp.render(this.canvas.getContext('2d') as CanvasRenderingContext2D);
     }
 
-    constructor(sp: DG.ScatterPlotViewer, xAxis: string, yAxis: string, lines: ILineOpts[]) {
+    constructor(sp: DG.ScatterPlotViewer, xAxis: string, yAxis: string, lines: ILines) {
         this.sp = sp;
-        this.xAxis = xAxis;
-        this.yAxis = yAxis;
+        this.xAxisValues = this.sp.dataFrame!.columns.byName(xAxis).toList();
```

Comment:
Why do we need it?

Reply:
I got it. This is TERRIBLY inefficient. We don't need it. You just doubled the memory amount we use for storing x,y numbers.

Reply:
Removed